### PR TITLE
plasma5Packages.ksysguard: remove if !allowAliases

### PR DESCRIPTION
**Pull request by alyssais**
###### Motivation for this change

Usually in Nixpkgs, attributes that are remnants of old packages are only included when aliases are enabled (which they are by default). This brings plasma5Packages into line with that, and also makes it easier to check the whole thing evaluates when working on the package set.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/plasma-5/default.nix [hunk 1]
```diff
@@ -164,6 +162,8 @@ let
         parachute = callPackage ./3rdparty/kwin/scripts/parachute.nix { };
       };
 
+    } // lib.optionalAttrs (config.allowAliases or true) {
+      ksysguard = throw "ksysguard has been replaced with plasma-systemmonitor";
```
Comment (ttuegel) [inline pkgs/desktops/plasma-5/default.nix:166]: I'm confused about this: won't this cause an error to be thrown when `allowAliases = true`?

Comment (alyssais) [reply pkgs/desktops/plasma-5/default.nix:166]: Yes, exactly like how other such warnings work -- compare to pkgs.ytop, as a randomly chosen example.

When aliases are enabled, we produce errors, when they're disabled, the attribute just doesn't exist.

Comment (ttuegel) [reply pkgs/desktops/plasma-5/default.nix:166]: That the name "allow" means "throw an error" is worse than useless, but that's not your fault. :frowning:

**Comment by Ma27** [comment]
> That the name "allow" means "throw an error" is worse than useless, but that's not your fault. :frowning:

To be fair, this applies to all aliases (with most of them defined in `pkgs/top-level/aliases.nix`) where a lot of them point to a different attribute (i.e. a backwards-compat alias for a renamed package attribute), however some aliases indicate a full removal where an error is thrown.

What I'm trying to say is: yes, the naming is in this case unfortunate, but since most of the aliases point to a different package, there's a reason behind the naming :)

**Pull request merged**